static/shared/js/typeahead.js

```diff
@@ -132,7 +134,9 @@ export function triage(query, objs, get_item) {
     for (const obj of objs) {
         const item = get_item(obj);
 
-        if (item.startsWith(query)) {
+        if (item === query) {
```

Comment:
Do we want this check to be case-insensitive?  I suppose probably "most consistent" would be to add case-sensitive exact matches before case-insensitive exact matches, but I think users will expect `jo` to sort "Jo` before "Joan" as well.